scripts/inline_comments_data_collection.py

```diff
@@ -47,16 +47,43 @@ def find_revision_from_patch(patch_id):
     return revision_phid
 
 
-def find_transactions_from_patch(patch_id):
+def create_index(file_path, index_path):
+    index = {}
+    offset = 0
+
+    with open(file_path, "rb") as file:
+        for line in file:
+            try:
+                entry = orjson.loads(line)
+                phid = entry.get("phid")
+                if phid:
+                    index[phid] = offset
+            except orjson.JSONDecodeError:
+                print(f"Error decoding JSON: {line}")
+            offset = file.tell()
```

Comment:
This is a bit brittle because the order of the revisions could change in the revisions.json file, rendering the index file no longer usable.

It's also an optimization that maybe is not useful, given the memory requirements are not so high. You could just read the entire revisions.json file in memory into a map.

Another option could be to change `get_all_inline_comments' to also return the revision in addition to the diff_id and comments, and then you could get the transactions from this revision object.
You'd simply need to change https://github.com/mozilla/bugbug/blob/dba47657d00fffdb185e0d69c980e5800b75e785/bugbug/tools/code_review.py#L587 to `yield revision, diff_id, comments` (and do th

Reply:
Implemented the first suggestion here (using a map): a5d1320ee2ba99589f56b03f060bb6d2d94fbe2b